rcl/test/memory_tools/memory_tools_common.cpp

```diff
@@ -65,9 +65,11 @@ custom_malloc(size_t size)
   }
   void * memory = malloc(size);
   uint64_t fw_size = size;
-  MALLOC_PRINTF(
-    " malloc (%s) %p %" PRIu64 "\n",
-    malloc_expected ? "    expected" : "not expected", memory, fw_size);
+  if (!malloc_expected) {
+    MALLOC_PRINTF(
+      " malloc (%s) %p %" PRIu64 "\n",
+      malloc_expected ? "    expected" : "not expected", memory, fw_size);
```

Comment:
I want to leave that for now, as I didn't always have the `if` statement around it and it's useful to remove it temporarily to see mallocs that are happening outside of the critical section.